bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/graphics/FontData.java

```diff
@@ -186,6 +186,26 @@ public FontData(String string) {
 	}
 }
 
+/**
+ * Constructs a new FontData copy
+ *
+ * @param fontData the FondData object for which copy is needed to be made
+ *
+ * @exception IllegalArgumentException
+ * <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the argument is null</li>
+ * </ul>
+ * @since 3.130
+ */
+public FontData(FontData fontData) {
```

Comment:
Same as for the other implementations: why not simply copy the fields from the original font data instead of using setters with additional checks that may not be useful when already having a properly initialized font data hand? In addition, some data like lang, country, and variant seem to not be copied by the copy constructors.